xe-daemon/xe-daemon.go

```diff
@@ -68,6 +72,20 @@ func main() {
 		return
 	}
 
+	// Reset pm_freeze_timeout to default value every time the daemon starts
+	if err := ioutil.WriteFile(SysFreezeTimeoutPath, []byte(DefaultTimeout), 0644); err != nil {
+		logger.Printf("Reset pm freeze timeout failed: %v", err)
+	}
+
+	watchChannel, watch_err := xs.Watch([]string{SuspendStatusPath})
+	if watch_err != nil {
```

Comment:
Do we need to retry when watch failed?

Reply:
it is not necessary, even watch failed, it will not block the following process and VM suspending. and typically, according to the implement of xapi, we do not retry for the operation of xenstore-write and watch.